### PR TITLE
gRPC: make @Transactional methods and classes blocking

### LINKED ISSUE #19580: Treat `@Transactional` as `@Blocking` for gRPC
### Description

Treat `@Transactional` as `@Blocking` for gRPC

### Implementation ideas

_No response_

### PR DESCRIPTION
**Pull request by michalszynkiewicz**
fixes #19580

### CODE REVIEW

File: extensions/grpc/deployment/src/main/java/io/quarkus/grpc/deployment/GrpcServerProcessor.java [hunk 1]
```diff
@@ -219,13 +222,16 @@ void discoverBindableServices(BuildProducer<BindableServiceBuildItem> bindables,
 
     private Set<MethodInfo> gatherBlockingMethods(ClassInfo service) {
         Set<MethodInfo> result = new HashSet<>();
-        boolean classHasBlocking = service.classAnnotation(GrpcDotNames.BLOCKING) != null;
+        // if we have @Transactional, and no @NonBlocking,
+        boolean blockingClass = service.classAnnotation(BLOCKING) != null
+                || (service.classAnnotation(TRANSACTIONAL) != null &&
+                        service.classAnnotation(NON_BLOCKING) == null);
```
Comment (cescoffier) [inline extensions/grpc/deployment/src/main/java/io/quarkus/grpc/deployment/GrpcServerProcessor.java:228]: Something we should report, and not only in gRPC is when @Blocking and @NonBlocking are used at the same time on the same element. It's clearly illegal, and should be reported properly.

CC @geoand

File: extensions/grpc/deployment/src/test/java/io/quarkus/grpc/server/blocking/TransactionalAsBlockingTest.java [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import com.dam.Blocking.ThreadName;
+import com.dam.Blocking1;
+import com.dam.Blocking2;
+import com.dam.Blocking3;
+import com.dam.MutinyBlocking2Grpc;
+import com.dam.MutinyBlocking3Grpc;
```
Comment (cescoffier) [inline extensions/grpc/deployment/src/test/java/io/quarkus/grpc/server/blocking/TransactionalAsBlockingTest.java:19]: For my own curiosity, why com.dam ?

**Review by cescoffier — CHANGES_REQUESTED** [review]

File: extensions/grpc/deployment/src/main/java/io/quarkus/grpc/deployment/GrpcServerProcessor.java (hunk 1 above)
Comment (geoand) [reply extensions/grpc/deployment/src/main/java/io/quarkus/grpc/deployment/GrpcServerProcessor.java:228]: I think we do that already in RR, but I might be misremembering.
I'll check tomorrow and add it if we don't

File: extensions/grpc/deployment/src/test/java/io/quarkus/grpc/server/blocking/TransactionalAsBlockingTest.java (hunk 2 above)
Comment (michalszynkiewicz) [reply extensions/grpc/deployment/src/test/java/io/quarkus/grpc/server/blocking/TransactionalAsBlockingTest.java:19]: I don't remember...

File: extensions/grpc/deployment/src/main/java/io/quarkus/grpc/deployment/GrpcServerProcessor.java (hunk 1 above)
Comment (geoand) [reply extensions/grpc/deployment/src/main/java/io/quarkus/grpc/deployment/GrpcServerProcessor.java:228]: Fixed in https://github.com/quarkusio/quarkus/pull/19601

Comment (michalszynkiewicz) [reply extensions/grpc/deployment/src/main/java/io/quarkus/grpc/deployment/GrpcServerProcessor.java]: I think it should be handled somewhere globally but Georgios does not so I added checks here ;)

**Review by cescoffier — APPROVED** [review]

**Pull request merged**